site/docs/bazel-and-java.md

```diff
@@ -179,6 +202,9 @@ a `BUILD` file and registering it either by adding `register_toolchain` rule to
 the `WORKSPACE` file or by using
 [`--extra_toolchains`](user-manual.html#flag--extra_toolchains) flag.
```

Comment:
I realize this predates your change and this doesn't need to be resolved now, but I think it would be valuable to have a more complete example here (or linked from here) that shows defining and registering a custom toolchain. I tried to remember how to set up a 'vanilla' toolchain recently and couldn't immediately remember how to register it, so having complete documentation for that might be helpful for end users.

Reply:
This should be as simple now as:
```
default_java_toolchain(
  name = "vanilla",
  configuration = VANILLA_TOOLCHAIN_CONFIGURATION,
  java_runtime = "@local_jdk//:local"
)
```
and 
`--extra_toolchains=//:vanilla_definition`

I didn't add a specific example for this, because everything is already there. Just need to put it together. Perhaps I can add it in a new PR.

Reply:
Is `:vanilla_definition` guaranteed to 'win' over the other built-in toolchains if they're both registered? I thought it might be necessary to add a constraints that isn't present on the built-in ones.

Reply:
Short answer, yes if --java_language_version matches the one in the toolchain (this is in the document). 
The definition is only constrained by the version.
For toolchains order of registration is important: first flags, then WORKSPACE, then jdk.WORKSPACE suffix. So user defined stuff always comes before built-in toolchains.